File: Backend/AstroGame.Shared/Models/Resources/InputResource.cs
using System;

namespace AstroGame.Shared.Models.Resources
{
    public class InputResource
    {
        public Guid Id { get; set; }

        /// <summary>
        /// The Resource that needs to be converted
        /// </summary>
        public Guid ResourceId { get; set; }

        public Guid OutputMaterialId { get; set; }

        /// <summary>
        /// The amount which is used
        /// </summary>
        public double InputValue { get; set; }

        /// <summary>
        /// The Resource that needs to be converted
        /// </summary>
        public virtual Resource Input { get; set; }

        public virtual ResourceManufaction Output { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/resources/input_resource.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/resources/resource_manufaction.dart';
import 'package:flutter_guid/flutter_guid.dart';
import 'package:json_annotation/json_annotation.dart';

part 'input_resource.g.dart';

@GuidConverter()
@JsonSerializable()
class InputResource {
  @JsonKey()
  Guid id;

  @JsonKey()
  Guid resourceId;

  @JsonKey()
  Guid outputMaterialId;

  @JsonKey()
  double inputValue;

  //@JsonKey()
  //Resource input;

  @JsonKey()
  ResourceManufaction output;

  InputResource(
    this.id,
    this.resourceId,
    this.outputMaterialId,
    this.inputValue,
    //this.input,
    this.output,
  );

  factory InputResource.fromJson(Map<String, dynamic> json) =>
      _$InputResourceFromJson(json);
  Map<String, dynamic> toJson() => _$InputResourceToJson(this);
}
